criticality_score/run.py

```diff
@@ -528,23 +539,30 @@ def get_repository(url):
     repo_url = parsed_url.path.strip('/')
     if parsed_url.netloc.endswith('github.com'):
         repo = None
+        last_commit = None
         try:
             repo = get_github_auth_token().get_repo(repo_url)
+            last_commit = repo.get_commits()[0]
```

Comment:
What i meant was calculate last_commit time inside get_repository_stats and bailout there. you can do it as a first call to self.last_commit

def get_repository_stats(repo, additional_params=None):
    if not repo.last_commit:
      return None

and then define 
def last_commit property(self):
   if self._last_commit:
     return self._last_commit
   self._last_commit = self.get_commit()[0]
  return  self._last_commit

No need to calculate it here, it is better to do this calculation inside the class function itself.